R/pkg/inst/tests/test_sparkSQL.R

```diff
@@ -949,6 +949,17 @@ test_that("fillna() on a DataFrame", {
   expect_identical(expected, actual)
 })
 
+test_that("crosstab() on a DataFrame", {
+  rdd <- lapply(parallelize(sc, 0:5), function(x) {
+    list(paste0("a", x %% 3), paste0("b", x %% 2))
+  })
+  df <- toDF(rdd, list("a", "b"))
+  ct <- crosstab(df, "a", "b")
+  ordered <- ct[order("a_b"),]
+  expected <- data.frame("a_b" = c("a0", "a1", "a2"), "b0" = c(1, 1, 1), "b1" = c(1, 1, 1))
```

Comment:
I think I figured out what is going on here -- The `expected` data.frame creates the column `a_b` as factors. You can pass in `stringsAsFactors=F` to the data.frame constructor to avoid this.

Also I think the `order` command above should probably be `ordered <- ct[order(ct$a_b),]` to get all the rows back out of it. It still associates row names which makes it not identical (you can do `row.names(ordered) <- NULL`).
A simpler check might be to just get one or two rows from `ct` and then compare them with `expected`